packages/plugin-react/src/react.ts

```diff
@@ -52,3 +52,28 @@ export const applyBasicReactSupport = (
       .use(ReactRefreshRspackPlugin, [{ include: [SCRIPT_REGEX] }]);
   });
 };
+
+export const applyReactProfiler = (api: RsbuildPluginAPI) => {
+  api.modifyRsbuildConfig((config, { mergeRsbuildConfig }) => {
+    const enableProfilerConfig: RsbuildConfig = {
+      output: {
+        minify: {
+          jsOptions: {
+            // Need to keep classnames and function names like Components for debugging purposes.
+            mangle: {
+              keep_classnames: true,
+              keep_fnames: true,
+            },
+          },
+        },
+      },
+    };
+    return mergeRsbuildConfig(config, enableProfilerConfig);
+  });
+
+  api.modifyBundlerChain((chain) => {
+    // Replace react-dom with the profiling version.
+    // Reference: https://gist.github.com/bvaughn/25e6233aeb1b4f0cdb8d8366e54a3977
+    chain.resolve.alias.set('react-dom$', 'react-dom/profiling');
```

Comment:
Should we alias 'scheduler/tracing' to 'scheduler/tracing-profiling' like CRA?

https://github.com/facebook/create-react-app/blob/0a827f69ab0d2ee3871ba9b71350031d8a81b7ae/packages/react-scripts/config/webpack.config.js#L329

https://gist.github.com/bvaughn/25e6233aeb1b4f0cdb8d8366e54a3977?permalink_comment_id=4409927#gistcomment-4409927

Reply:
I tested it without scheduler alias against v17-v18 and its working without any issues. But for assurance, we can add the alias.

Reply:
Yes, there are still a small number of users using React 16, adding the alias can be better😄